### PR TITLE
Make possible to specify a working directory

### PR DESCRIPTION
**Pull request by widoz**
There might be edge cases where a project might have different modules or packages requiring a specific configuration to build the assets. In most cases the problem is solvable with monorepo strategies but it's not always possible due to a different set of reasons, mainly legacy reasons and different unresolvable dependencies hierarchies.

Github allow to specify a `workflow-directory` different than the project root which would help in such cases.

The changes are non intrusives and are an opt-in enhancement, hence they will not force any consumer to adapt to the new configuration.

### CODE REVIEW

File: .github/workflows/build-and-push-assets.yml [hunk 1]
```diff
@@ -3,6 +3,11 @@ name: Build and push assets
 on:
   workflow_call:
     inputs:
+      WORKING_DIRECTORY:
+        description: The Working Directory.
```
Comment (tyrann0us) [inline .github/workflows/build-and-push-assets.yml:7]:
```suggestion
        description: Working directory path.
```

**Review by tyrann0us — COMMENTED** [review]
Thanks for this! Please allow me two comments:
1. Do you think having the work dir configurable in all available workflows makes sense? If so, it would be really great if you could add it everywhere.
2. Please update the documentation.

Thanks!

**Comment by gmazzap** [comment]
@tyrann0us looking at other workflows, I can't think of a reason why it would be necessary to change the working dir.

I've committed your suggestion and updated the docs.

If we can merge this morning, it would be really helpful because it is a blocker for us. Otherwise, we'll need to require the PR's source branch to unblock our work.

**Review by gmazzap — APPROVED** [review]

**Review by tyrann0us — APPROVED** [review]
> If we can merge this morning, it would be really helpful because it is a blocker for us.

Understood. Merging now.

**Comment by widoz** [comment]
@gmazzap @tyrann0us Thanks a bunch for the corrections.

> Do you think having the work dir configurable in all available workflows makes sense? If so, it would be really great if you could add it everywhere.

Honestly I don't know, for the assets I had a real case scenario but unfortunately I do not use much the reusable workflows yet and I don't know. For sure it would be a flexibility that does not produce side effects, so why not? 🤷🏽 

That being said, for PHPCS for instance I found difficult to have a different working directory and probably it highly depends on how we organize our code. 

In conclusion if it would be necessary we can still add it, we know this feature exists.

Right now I really need this so I would merge it.
